controllers/base_controller.js

```diff
@@ -125,7 +123,15 @@ module.exports = function BaseControllerModule(pb) {
             self.templateService.load('admin/elements/wysiwyg', function(err, data) {
                 cb(err, new pb.TemplateValue(data, false));
             });
-        });
+        });      
+        var analyticsTags = pb.AnalyticsManager.getTemplateTags();
+        if(analyticsTags.length > 0){
+          analyticsTags.forEach(function(result){
+            self.templateService.registerLocal(result.templateTag, function(flag, cb){
+              pb.AnalyticsManager.onPageRender(self.req, self.session, self.ls, result.providerName, cb);
+            });
+          });
```

Comment:
This does avoid the performance issues I mentioned but brings to light another issue with this implementation.  It does not support the ability for multiple providers to be rendered under the same analytics tag.  This is needed for backward compatibility.

For example, if someone were running both the google analytics plugin and the ChartBeat plugin only one would get rendered when the ```^analytics^``` flag is encountered in the templates.  The reason is because in your loop you register a single flag name with a single provider.  

The more I ponder over this use case the more I wonder if the core functionality should be to provide an ```analytics_header``` and ```analytics_footer``` flag and let the plugins themselves choose where to render with the default going to the header.  Then if there are other specific use cases for inputing in the body or only in certain circumstances the plugins can register global flags and render appropriately for that specific purpose.